acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -43,6 +43,29 @@ def expect_file_on_host_to_contain(host, file, expected, seconds=30)
   end
 end
 
+# Show the logs of the broker and agents - useful to see why a test failed
+def show_pcp_logs
+  puts "---- Broker log -----"
+  on(master, "cat /var/log/pcp-broker.log") do |result|
+    puts result.stdout
+  end
+
+  agents.each do |agent|
+    puts "----- agent #{agent} log -----"
+    on(agent, "cat #{logfile(agent)}") do |result|
+      puts result.stdout
+    end
+  end
+end
+
+# Evaluate the block, show logs if test assertions were triggered
+def show_pcp_logs_on_failure(&block)
+  yield
+rescue MiniTest::Assertion => exception
+  show_pcp_logs
```

Comment:
I tested out a deliberate failure and got output like this:

```
C93807 - Associate pxp-agent with a PCP broker
localhost $ scp /tmp/beaker20160413-25826-emv3w8 l7vlvi75pwdnrz2.delivery.puppetlabs.net:/cygdrive/c/ProgramData/PuppetLabs/pxp-agent/etc/pxp-agent.conf {:ignore => }

* Stop pxp-agent if it is currently running

* Assert that the agent is not listed in pcp-broker inventory

* Start pxp-agent service

* Assert that agent is listed in pcp-broker inventory
---- Broker log -----
2016-04-13 04:20:59,371 INFO  [async-dispatch-2] [p.t.s.w.jetty9-service] Initializing web server(s).
2016-04-13 04:20:59,409 INFO  [async-dispatch-2] [p.p.b.service] Initializing broker service
```

If anyone outside the current core team was to encounter this in Jenkins output; it would be helpful if it was prefaced with a log/puts message like "Assertion exception was caught - now outputting pcp-broker and pxp-agent logs" so that the appearance of a pcp-broker log in the output is explained